flappy_bird_example/metal_owl.py

```diff
@@ -0,0 +1,227 @@
+class Ground(Sprite):
+    def update(self):
+        self.x -= 2
+        if self.x < -24:
+            self.x += 24
+
+# the game is called as a widget
+# the background is set and added as a widget
+# the pipes are added as a widget under the ground so that the ground covers bottom edge
+# the ground is added as a widget
+# the score label is centered but at the top of the screen with a default text of "0" score
+# the over label of "Game Over" text has it's opacity set to 0 so is invisible
+# the bird player is set (x, y) = (20, half of screen) and added as a widget
+# the clock calls for updates at 60 times per second
+# game_over is default set to false and the score is default set to 0
+class Game(Widget):
+    def __init__(self):
+        super(Game, self).__init__()
+        self.background = Background(source="Resources/kivy-game-dev/flappy/images/background.png")
+        self.size = self.background.size
+        self.add_widget(self.background)
+        self.ground = Ground(source="Resources/kivy-game-dev/flappy/images/ground.png")
+        self.pipes = Pipes(pos=(0, self.ground.height), size=self.size)
+        self.add_widget(self.pipes)
+        self.add_widget(self.ground)
+        self.score_label = Label(center_x=self.center_x,
+            top=self.top - 30, text="0")
+        self.add_widget(self.score_label)
+        self.over_label = Label(center=self.center, opacity=0,
+            text="Game Over")
+        self.add_widget(self.over_label)
+        self.bird = Bird(pos=(20, self.height / 2))
+        self.add_widget(self.bird)
+        Clock.schedule_interval(self.update, 1.0/60.0)
+        self.game_over = False
+        self.grounded = False
+        self.score = 0
+
+# update by delta time
+# check for game_over
+# check for bird collision with ground and change game_over accordingly
+# check for bird collision with top and bottom pipe and change game_over accordingly
+# if the position of the pipe is behind the player, set score +1 and set scored to true
+# set the score label to reflect the string score and play score sound
+    def update(self, dt):
+        if self.game_over:
+            return
+
+        if self.grounded:
+            return
+
+        self.background.update()
+        self.bird.update()
+        self.ground.update()
+        self.pipes.update(dt)
+
+        if self.bird.collide_widget(self.ground):
+            self.grounded = True
+            self.velocity_y = 0
+
+        #if self.bird.collide_widget(self.ground):
+            #self.game_over = True
+
+        for pipe in self.pipes.children:
+            if pipe.top_image.collide_widget(self.bird):
+                self.game_over = True
+            elif pipe.bottom_image.collide_widget(self.bird):
+                self.game_over = True
+            elif not pipe.scored and pipe.right < self.bird.x:
+                pipe.scored = True
+                self.score += 1
+                self.score_label.text = str(self.score)
```

Comment:
You could use a Kivy [NumericProperty](https://kivy.org/docs/api-kivy.properties.html#kivy.properties.NumericProperty) for the score to remove the need to manually update the label text. There are probably other situations you could use Kivy properties too.